packages/channel-client/src/channel-client.ts

```diff
@@ -32,7 +34,12 @@ export class ChannelClient implements ChannelClientInterface {
     return this.provider.walletVersion;
   }
 
-  constructor(readonly provider: ChannelProviderInterface) {}
+  constructor(readonly provider: ChannelProviderInterface) {
+    this.channelState = new ReplaySubject(1);
```

Comment:
Are you sure that a ReplaySubject is needed? Is the latest event delivered to an observer on subscription without a ReplaySubject?

Reply:
> Is the latest event delivered to an observer on subscription without a ReplaySubject?

It's not, that's why I chose to use ReplaySubject instead of a normal Subject. It ensures that it will always emit the last state of the channel.

Reply:
Yep, and I wonder if a `Subject` is equivalent to `ReplaySubject(1)`. I might be wrong though

Reply:
I think `ReplaySubject(1)` is equivalent to `BehaviourSubject`, but not `Subject`

Reply:
> Yep, and I wonder if a `Subject` is equivalent to `ReplaySubject(1)`. I might be wrong though

It's not, it doesn't save the last emitted value.: https://jsbin.com/yekaqov/edit?js,console


> I think `ReplaySubject(1)` is equivalent to `BehaviourSubject`, but not `Subject`

When the replaySubject constructor param is 1, is equivalent to BehaviourSubject, but it doesn't need an initial value.